### PR TITLE
B2b account

### PR DESCRIPTION
**Pull request by mjasrotia**
1. “account.schema.json” – This is the account class. Have only few account attributes which are immutable.
2. “account-crm.schema.json” – This is the account CRM mixin which has the standard set of Account attributes.
3. “account-marketo”  - This is a mixin which has the attributes specific to the MARKETO system. It has two nested properties.
-   “market:org” – Attributes from “mkt_org” entity.
-  “market:namedAccount” – Attributes from “mkt_account” entity.

### CODE REVIEW

File: extensions/adobe/experience/marketo/account-marketo.schema.json [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            },
+            "marketo:additionalFields": {
+              "title": "Additional Fields",
+              "description": "Additional fields.",
+              "type": "string"
+            }
+          },
+          "marketo:namedAccount": {
+            "properties": {
+              "marketo:opptyAmount": {
```
Comment (cfraser) [inline extensions/adobe/experience/marketo/account-marketo.schema.json:60]: I'd spell out `opportunity` or drop `oppty`

File: extensions/adobe/experience/marketo/account-marketo.schema.json [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                "title": "Opportunity Amount",
+                "description": "Opportunity amount.",
+                "type": "number"
+              },
+              "marketo:opptyCount": {
```
Comment (cfraser) [inline extensions/adobe/experience/marketo/account-marketo.schema.json:65]: Spell out `opportunity` here or drop `oppty`

File: extensions/adobe/experience/marketo/account-marketo.schema.json [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                "items": {
+                  "type": "number"
+                }
+              },
+              "marketo:membershipCount": {
+                "title": "Membership Count",
+                "description": "Membership count.",
+                "type": "number"
+              },
+              "marketo:CRMOrgID": {
```
Comment (cfraser) [inline extensions/adobe/experience/marketo/account-marketo.schema.json:91]: `CRM` should be lower case.

File: schemas/context/account-crm.schema.json [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        },
+        "xdm:fax": {
+          "title": "Fax Number",
+          "description": "Fax number.",
+          "type": "array",
+          "items": {
+            "$ref": "https://ns.adobe.com/xdm/context/phonenumber"
+          }
+        },
+        "xdm:account-organization": {
```
Comment (cfraser) [inline schemas/context/account-crm.schema.json:75]: `accountOrganization` 

Hyphens are discouraged.

**Comment by cfraser** [comment]
Looks good besides some style tweaks.

File: schemas/context/account-crm.schema.json (hunk 4 above)
Comment (mjasrotia) [reply schemas/context/account-crm.schema.json]: Resolved

File: extensions/adobe/experience/marketo/account-marketo.schema.json (hunk 3 above)
Comment (mjasrotia) [reply extensions/adobe/experience/marketo/account-marketo.schema.json]: Resolved

File: extensions/adobe/experience/marketo/account-marketo.schema.json (hunk 2 above)
Comment (mjasrotia) [reply extensions/adobe/experience/marketo/account-marketo.schema.json]: Resolved

File: extensions/adobe/experience/marketo/account-marketo.schema.json (hunk 1 above)
Comment (mjasrotia) [reply extensions/adobe/experience/marketo/account-marketo.schema.json]: Resolved

File: schemas/context/account.schema.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:extends": [
+    "https://ns.adobe.com/xdm/data/record"
+  ],
+  "description": "An XDM Business Account.",
+  "definitions": {
+    "account": {
+      "properties": {
+        "xdm:accountID": {
```
Comment (khanna-vikas) [inline schemas/context/account.schema.json:22]: accountID is a required attribute.

File: schemas/context/account-crm.schema.json [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+          "title": "Parent Account",
+          "description": "Parent company for companies that are subsidiaries of a larger company or organization. The parent account must be an existing account in salesforce. You can enter the account name, or select (or optionally, create) the account using the lookup icon.",
+          "type": "string"
+        },
+        "xdm:partnerAccountID": {
+          "title": "Partner Account Id",
+          "description": "Partner account id.",
+          "type": "string"
+        },
+        "xdm:masterRecordID": {
```
Comment (khanna-vikas) [inline schemas/context/account-crm.schema.json:37]: What's the significance of MasterRecordID?

File: schemas/context/account-crm.example.1.json [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+      "@id": "https://data.adobe.io/entities/address/234",
+      "xdm:primary": true,
+      "xdm:street1": "100 Park Ave",
+      "xdm:city": "San Jose",
+      "xdm:stateProvince": "US-CA",
+      "xdm:postalCode": "95110",
+      "xdm:country": "United States",
+      "xdm:countryCode": "US",
+      "schema:latitude": 37.3382,
+      "schema:longitude": 121.8863,
```
Comment (khanna-vikas) [inline schemas/context/account-crm.example.1.json:90]: -121.8863

File: schemas/context/account-crm.example.1.json [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+      "xdm:status": "active",
+      "xdm:lastVerifiedDate": "2018-01-02"
```
Comment (khanna-vikas) [inline schemas/context/account-crm.example.1.json:92]: Make "dd" differentiate from the month (> 12)

File: schemas/context/account-crm.example.1.json [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+      "xdm:street1": "345 Park Ave",
+      "xdm:city": "San Jose",
+      "xdm:stateProvince": "US-CA",
+      "xdm:postalCode": "95110",
+      "xdm:country": "United States",
+      "xdm:countryCode": "US",
+      "schema:latitude": 37.3382,
+      "schema:longitude": 121.8863,
+      "xdm:status": "active",
+      "xdm:lastVerifiedDate": "2018-01-02"
```
Comment (khanna-vikas) [inline schemas/context/account-crm.example.1.json:76]: same here

File: schemas/context/account-crm.example.1.json [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+      "@id": "https://data.adobe.io/entities/address/123",
+      "xdm:primary": true,
+      "xdm:street1": "345 Park Ave",
+      "xdm:city": "San Jose",
+      "xdm:stateProvince": "US-CA",
+      "xdm:postalCode": "95110",
+      "xdm:country": "United States",
+      "xdm:countryCode": "US",
+      "schema:latitude": 37.3382,
+      "schema:longitude": 121.8863,
```
Comment (khanna-vikas) [inline schemas/context/account-crm.example.1.json:74]: same as below

File: schemas/context/account-crm.example.1.json [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+        "xdm:street1": "100 Lincoln Ave",
+        "xdm:city": "San Jose",
+        "xdm:stateProvince": "US-CA",
+        "xdm:postalCode": "95125",
+        "xdm:country": "United States",
+        "xdm:countryCode": "US",
+        "schema:latitude": 37.3382,
+        "schema:longitude": 121.8863,
+        "xdm:status": "active",
+        "xdm:lastVerifiedDate": "2018-01-03"
```
Comment (khanna-vikas) [inline schemas/context/account-crm.example.1.json:41]: modify the "dd" (> 12) to differentiate mm

File: schemas/context/account-crm.example.1.json [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+        "xdm:primary": true,
+        "xdm:type": "work",
+        "xdm:street1": "100 Lincoln Ave",
+        "xdm:city": "San Jose",
+        "xdm:stateProvince": "US-CA",
+        "xdm:postalCode": "95125",
+        "xdm:country": "United States",
+        "xdm:countryCode": "US",
+        "schema:latitude": 37.3382,
+        "schema:longitude": 121.8863,
```
Comment (khanna-vikas) [inline schemas/context/account-crm.example.1.json:39]: -121.8863

File: schemas/context/account-crm.example.1.json [hunk 13]
```diff
@@ -0,0 +1,95 @@
+{
+  "xdm:ownerID": "OWNER111",
+  "xdm:parentID": "ACC000",
+  "xdm:partnerAccountID": "PRTNR1111",
+  "xdm:masterRecordID": "MSTR111",
```
Comment (khanna-vikas) [inline schemas/context/account-crm.example.1.json:5]: Whats the importance?

File: extensions/adobe/experience/marketo/account-marketo.schema.json [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/account"],
+  "meta:extends": ["https://ns.adobe.com/xdm/context/account-crm"],
+  "description": "Marketo specific Account fields.",
+  "definitions": {
+    "account-marketo": {
+      "properties": {
+        "marketo:marketoID": {
```
Comment (khanna-vikas) [inline extensions/adobe/experience/marketo/account-marketo.schema.json:20]: Should be marketo:accountID

Comment (khanna-vikas) [reply extensions/adobe/experience/marketo/account-marketo.schema.json:20]: Also should be required.

File: extensions/adobe/experience/marketo/account-marketo.example.1.json [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+      "@id": "https://data.adobe.io/entities/address/234",
+      "xdm:primary": true,
+      "xdm:street1": "100 Park Ave",
+      "xdm:city": "San Jose",
+      "xdm:stateProvince": "US-CA",
+      "xdm:postalCode": "95110",
+      "xdm:country": "United States",
+      "xdm:countryCode": "US",
+      "schema:latitude": 37.3382,
+      "schema:longitude": 121.8863,
```
Comment (khanna-vikas) [inline extensions/adobe/experience/marketo/account-marketo.example.1.json:116]: check?

File: extensions/adobe/experience/marketo/account-marketo.example.1.json [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+      "xdm:status": "active",
+      "xdm:lastVerifiedDate": "2018-01-02"
```
Comment (khanna-vikas) [inline extensions/adobe/experience/marketo/account-marketo.example.1.json:118]: Modify the value.

File: extensions/adobe/experience/marketo/account-marketo.example.1.json [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+      "xdm:street1": "345 Park Ave",
+      "xdm:city": "San Jose",
+      "xdm:stateProvince": "US-CA",
+      "xdm:postalCode": "95110",
+      "xdm:country": "United States",
+      "xdm:countryCode": "US",
+      "schema:latitude": 37.3382,
+      "schema:longitude": 121.8863,
+      "xdm:status": "active",
+      "xdm:lastVerifiedDate": "2018-01-02"
```
Comment (khanna-vikas) [inline extensions/adobe/experience/marketo/account-marketo.example.1.json:102]: same

File: extensions/adobe/experience/marketo/account-marketo.example.1.json [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+      "@id": "https://data.adobe.io/entities/address/123",
+      "xdm:primary": true,
+      "xdm:street1": "345 Park Ave",
+      "xdm:city": "San Jose",
+      "xdm:stateProvince": "US-CA",
+      "xdm:postalCode": "95110",
+      "xdm:country": "United States",
+      "xdm:countryCode": "US",
+      "schema:latitude": 37.3382,
+      "schema:longitude": 121.8863,
```
Comment (khanna-vikas) [inline extensions/adobe/experience/marketo/account-marketo.example.1.json:100]: - sign

File: extensions/adobe/experience/marketo/account-marketo.example.1.json [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+        "xdm:street1": "100 Lincoln Ave",
+        "xdm:city": "San Jose",
+        "xdm:stateProvince": "US-CA",
+        "xdm:postalCode": "95125",
+        "xdm:country": "United States",
+        "xdm:countryCode": "US",
+        "schema:latitude": 37.3382,
+        "schema:longitude": 121.8863,
+        "xdm:status": "active",
+        "xdm:lastVerifiedDate": "2018-01-03"
```
Comment (khanna-vikas) [inline extensions/adobe/experience/marketo/account-marketo.example.1.json:67]: differentiate "dd" from "mm"

File: extensions/adobe/experience/marketo/account-marketo.example.1.json [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+        "xdm:primary": true,
+        "xdm:type": "work",
+        "xdm:street1": "100 Lincoln Ave",
+        "xdm:city": "San Jose",
+        "xdm:stateProvince": "US-CA",
+        "xdm:postalCode": "95125",
+        "xdm:country": "United States",
+        "xdm:countryCode": "US",
+        "schema:latitude": 37.3382,
+        "schema:longitude": 121.8863,
```
Comment (khanna-vikas) [inline extensions/adobe/experience/marketo/account-marketo.example.1.json:65]: -121

File: extensions/adobe/experience/marketo/account-marketo.example.1.json [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+  "xdm:createdDate": "2018-01-02T15:52:25+00:00",
+  "xdm:lastUpdatedBy": "USER2222",
+  "xdm:lastUpdatedDate": "2018-01-02T15:52:25+00:00",
+  "xdm:lastActivityDate": "2018-01-02T15:52:25+00:00",
+  "xdm:lastReferencedDate": "2018-01-02T15:52:25+00:00",
+  "xdm:lastViewedDate": "2018-01-02T15:52:25+00:00",
+  "xdm:ownerID": "OWNER111",
+  "xdm:parentID": "ACC000",
+  "xdm:partnerAccountID": "PRTNR1111",
+  "xdm:masterRecordID": "MSTR111",
```
Comment (khanna-vikas) [inline extensions/adobe/experience/marketo/account-marketo.example.1.json:31]: Significance of this property?

File: extensions/adobe/experience/marketo/account-marketo.example.1.json [hunk 22]
```diff
@@ -0,0 +1,121 @@
+{
+  "marketo:marketoID": "MKTO1111",
```
Comment (khanna-vikas) [inline extensions/adobe/experience/marketo/account-marketo.example.1.json:2]: marketo:accountID

File: schemas/context/account-crm.example.1.json (hunk 7 above)
Comment (mjasrotia) [reply schemas/context/account-crm.example.1.json:90]: These examples were taken from existing Schema examples. Those need to be fixed as well. I will fix this one.

File: schemas/context/account-crm.example.1.json (hunk 8 above)
Comment (mjasrotia) [reply schemas/context/account-crm.example.1.json:92]: Please elaborate. I didn't understand. Could you please provide example of what is the concern here ?

File: schemas/context/account-crm.example.1.json (hunk 9 above)
Comment (mjasrotia) [reply schemas/context/account-crm.example.1.json:76]: Please elaborate. I didn't understand. Could you please provide example of what is the concern here ?

File: schemas/context/account-crm.example.1.json (hunk 10 above)
Comment (mjasrotia) [reply schemas/context/account-crm.example.1.json:74]: Will fix

File: schemas/context/account-crm.example.1.json (hunk 11 above)
Comment (mjasrotia) [reply schemas/context/account-crm.example.1.json:41]: These examples were taken from existing Schema examples. Those need to be fixed as well. I will fix this one.

File: schemas/context/account-crm.example.1.json (hunk 12 above)
Comment (mjasrotia) [reply schemas/context/account-crm.example.1.json:39]: These examples were taken from existing Schema examples. Those need to be fixed as well. I will fix this one.

File: extensions/adobe/experience/marketo/account-marketo.example.1.json (hunk 15 above)
Comment (mjasrotia) [reply extensions/adobe/experience/marketo/account-marketo.example.1.json:116]: These examples were taken from existing Schema examples. Those need to be fixed as well. I will fix this one.

File: schemas/context/account.schema.json (hunk 5 above)
Comment (mjasrotia) [reply schemas/context/account.schema.json:22]: Added the required

File: schemas/context/account-crm.schema.json (hunk 6 above)
Comment (mjasrotia) [reply schemas/context/account-crm.schema.json]: Removed

**Pull request merged**